packages/builder/builder-shared/src/fallback.ts

```diff
@@ -17,7 +17,7 @@ export const resourceRuleFallback = (
       // "..." refers to the webpack defaults
       rule === '...' ||
       // this is a special case, put the mjs fullySpecified rule in the outside
-      (rule.resolve && !rule.mimetype)
+      (rule.resolve && 'fullySpecified' in rule.resolve && !rule.mimetype)
```

Comment:
why need this

Reply:
When using `enableAssetFallback`, we need to put all rules inside the`oneof`, except the `fullySpecified` rule for `.mjs` (which is a special case)